src/OrleansProviders/Streams/Common/PooledCache/ICacheDataAdapter.cs

```diff
@@ -59,6 +60,24 @@ public interface ICacheDataAdapter<in TQueueMessage, TCachedMessage>
         /// Assignable purge action.  This is called when a purge request is triggered.
         /// </summary>
         Action<IDisposable> PurgeAction { set; }
+
+        /// <summary>
+        /// Handler for when a cachedMessage got purged
+        /// </summary>
+        /// <param name="cachedMessage"></param>
+        void SignalMessagePurged(ref TCachedMessage cachedMessage);
+
+        /// <summary>
+        /// Determine action to make on a purgeRequest
+        /// </summary>
+        /// <param name="purgeRequest"></param>
+        void ConsultPurgeRequest(IDisposable purgeRequest);
```

Comment:
it means, looking at this purgeRequest, make decision about it. The decision has two options currently, one is ignore it(the purge request is handled in other manner), second is dispose the purge request (which is dispose the block represented by the purge request)